core/components/atoms/avatar/AvatarContext.tsx

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+
+type ContextProps = {
+  size?: string;
+  appearance?: string;
+  firstName?: string;
+  lastName?: string;
+};
+
+const context = React.createContext<ContextProps>({
+  size: 'regular',
+  appearance: 'secondary',
+  firstName: '',
+  lastName: '',
+});
+
+export const AvatarProvider = context.Provider;
```

Comment:
Please consider exporting AvatarProvider as default and context as variable, as AvatarProvider is the component which manages the context, also rename the file to AvatarProvider.tsx for readability.